### ISSUE TITLE
Transactional flow is closing the producer only after 30s

### ISSUE THREAD

**Issue by gmarti** [akka/alpakka-kafka#989]
### Versions used 
akka-stream-kafka 1.1.0

tested with embedded-kafka 2.3.0 and local kafka container

### Expected Behavior
When stopping a transactional source + flow, the producer should be closed after finishing last transaction

### Actual Behavior
When stopping a transactional source+flow, the producer is closed 30s after the transaction is finished

### Relevant logs
We can see in the relevant logs that the transaction is finished at 14:10:36 with last log been : 
`Transition from state ABORTING_TRANSACTION to READY`
and we have 30s downtime before 
`Shutdown of Kafka producer I/O thread has completed.`

[logs](https://github.com/akka/alpakka-kafka/files/3901936/producer-30s-shutdown.log)

P.S maybe it's related to kafka client ?

**Comment by seglo** [comment]
Hi @gmarti.  This is likely related to the `stop-timeout` setting on the consumer, which is 30s by default.  We've discussed setting this configuration to `0` for transactional workloads or removing it altogether.  If you configure this to `0` does it resolve your issue?

https://doc.akka.io/docs/alpakka-kafka/current/consumer.html#settings

**Comment by gmarti** [comment]
I used `.withStopTimeout(Duration.Zero)` on the consumer settings
Can I check that it's correctly applied somehow ?
To me it seems it's getting locked because of the opened transaction, but i'm not sure.
The current behavior seems to open a new transaction just after committing even if it will not be used. Is there an "after transaction" timeout or something ?

**Comment by seglo** [comment]
> The current behavior seems to open a new transaction just after committing even if it will not be used. Is there an "after transaction" timeout or something ?

Yes, that's the current released behaviour.  When we merged the Kafka 2.4.0 support (#971) I noticed that `KafkaProducer.close` would block (for 60 seconds by default) when there are open transactions.  If we shutdown the stream (or if the stream completes) then in many cases the call to close the producer can block.  This didn't appear to be an issue with previous versions of the client, but we noticed it when updating to Kafka 2.4.0.

On a stream shutdown or cancellation signal we will now abort any open transactions before calling close.

https://github.com/akka/alpakka-kafka/commit/ada44cd9af68fb17bea710f1e17232f131194b97#diff-fd791f42a09d4f5533ef0b561f358d80R151-R171

**Comment by gmarti** [comment]
Thanks ! 

So next release candidate I'll make a test to see if it solves the problem for me
In the meantime I set a smaller producer closeTimeout that is doing the job

**Comment by ennru** [comment]
Did you have the chance to check if this was solved by the recent releases?
